### ISSUE TITLE
Mismatched help keys and task args doesn't result in a useful error

### ISSUE THREAD

**Issue by sohaibfarooqi** [pyinvoke/invoke#398]
Hi,
I am trying to add help text to a task as follow: (Pyinvoke 0.13.0)

```
@task(help={'name': "Use --type aj and --type mp"})
      def build(ctx, type=None):
         pass
```

The issue is invoke --help build prints nothing in `OPTIONS`.
The check `if name in self.help:` (Line: 186) inside function `def arg_opts()` in invoke/tasks.py is getting failed and not value gets assigned to `opts['help']` that's what i noticed. Any workaround for this issue?

**Comment by bitprophet** [comment]
Hm, IIRC the keys in `help` need to match the arguments in the task function itself, but your key is `name` while your argument is `type`.

Just checked, and my memory is correct - you need to change `'name'` in `help` to be `'type'`, and then `invoke --help build` should show your help text on the line for `--type`.

Gonna leave this open as "this should have resulted in a useful error instead of silent failure", however. Thanks for the report!

**Comment by bitprophet** [comment]
Fixed in #611.
